topics/python-programming/tutorials/10_functions/tutorial.md

```diff
@@ -183,12 +188,52 @@ There are several ways to solve this problem, however it might be easier to do i
 >    > print(hamming_distance(seq1,seq2))
 >    >  ```
 >    > </details>
+>    > <details markdown="1">
+>    > <summary>{% icon solution %} Solution 2
+>    > </summary>
+>    >
+>    >  ```python
+>    > # string1 and string2 should be the same length.
+>    > def hamming_distance(string1, string2): 
+>    >     """Return the Hamming distance between equal-length sequences."""
+>    >     
+>    >     if len(string1) != len(string2):
+>    >         raise ValueError("Undefined for sequences of unequal length.")
+>    >     
+>    >     # Start with a distance of zero, and count up
+>    >     distance = 0
+>    >     # Loop over the indices of the string
+>    >     for s1,s2 in zip(string1,string2):
+>    >         if s1 != s2:
+>    >              distance +=1
+>    >         return distance
+>    >     # Return the final count of differences
+>    >     return distance
+>    > 
+>    > seq1 = "GATCATAGA"
+>    > seq2 = "CATCATACA"
+>    > print(hamming_distance(seq1,seq2))
+>    >  ```
+>    > </details>
 >
 {: .hands_on}
 
 ---
 
+# with zip!
+def hamming_distance(string1, string2): 
+    """Return the Hamming distance between equal-length sequences."""
+    
+    if len(string1) != len(string2):
+        raise ValueError("Undefined for sequences of unequal length.")
+    
+    # Start with a distance of zero, and count up
+    distance = 0
+
```

Comment:
This seems to be missing some code?

Reply:
And the previous example is already using `zip()`

Reply:
Yes, it's too much there. THanks for noticing. Should be deleted.